### PR TITLE
OrgU: Repo Pattern for Positions and Authorities

### PR DESCRIPTION
**Pull request by tbongers-cat**
As proposed by @klees 
See commit message for more details.

### CODE REVIEW

**Comment by tbongers-cat** [comment]
This also includes fixes for [Mantis #35740](https://mantis.ilias.de/view.php?id=35740), [Mantis #35741](https://mantis.ilias.de/view.php?id=35741), [Mantis #35742](https://mantis.ilias.de/view.php?id=35742) and [Mantis #35746](https://mantis.ilias.de/view.php?id=35746)

**Comment by tbongers-cat** [comment]
Also includes a fix for [Mantis #35746](https://mantis.ilias.de/view.php?id=35746).

**Comment by tbongers-cat** [comment]
Added the small fix for [Mantis #34298](https://mantis.ilias.de/view.php?id=34928)

File: Modules/OrgUnit/classes/Positions/class.ilOrgUnitPositionFormGUI.php [hunk 1]
```diff
@@ -87,34 +127,54 @@ public function fillObject(): bool
...
-            $id = $authority["id"];
-            if ($id == '') {
-                $id = null;
-            }
-            $ilOrgUnitAuthority = ilOrgUnitAuthority::findOrGetInstance($id);
-            $ilOrgUnitAuthority->setPositionId($this->object->getId());
-            $ilOrgUnitAuthority->setScope($authority["scope"]);
-            $ilOrgUnitAuthority->setOver($authority["over"]);
-            $ilOrgUnitAuthorities[] = $ilOrgUnitAuthority;
+            $id = ($authority["id"] == '') ? null : $authority["id"];
```
Comment (nhaagen) [inline Modules/OrgUnit/classes/Positions/class.ilOrgUnitPositionFormGUI.php]:
```suggestion
            $id = ($authority["id"] == '') ? null : (int)$authority["id"];
```

**Review by nhaagen — APPROVED** [review]

File: Modules/OrgUnit/classes/Positions/class.ilOrgUnitPositionFormGUI.php (hunk 1 above)
Comment (tbongers-cat) [reply Modules/OrgUnit/classes/Positions/class.ilOrgUnitPositionFormGUI.php]: Done!

**Pull request merged**